src/main/drivers/timer.c

```diff
@@ -608,85 +610,85 @@ static void timCCxHandler(TIM_TypeDef *tim, timerConfig_t *timerConfig)
-    void name(void)                                                     \
+    IRQHANDLER(name)                                                    \
     {                                                                   \
-        timCCxHandler(TIM ## i, &timerConfig[TIMER_INDEX(i)]);          \
-        timCCxHandler(TIM ## j, &timerConfig[TIMER_INDEX(j)]);          \
+        if (USED_TIMERS & TIM_N(i))                                     \
+            timCCxHandler(TIM ## i, &timerConfig[TIMER_INDEX(i)]);      \
+        if (USED_TIMERS & TIM_N(j))                                     \
+            timCCxHandler(TIM ## j, &timerConfig[TIMER_INDEX(j)]);      \
     } struct dummy
 
 #define _TIM_IRQ_HANDLER(name, i)                                       \
-    void name(void)                                                     \
+    IRQHANDLER(name)                                                    \
     {                                                                   \
-        timCCxHandler(TIM ## i, &timerConfig[TIMER_INDEX(i)]);          \
+        if (USED_TIMERS & TIM_N(i))                                     \
+            timCCxHandler(TIM ## i, &timerConfig[TIMER_INDEX(i)]);      \
     } struct dummy
 
-#if USED_TIMERS & TIM_N(1)
-_TIM_IRQ_HANDLER(TIM1_CC_IRQHandler, 1);
-# if defined(STM32F10X)
-_TIM_IRQ_HANDLER(TIM1_UP_IRQHandler, 1);       // timer can't be shared
-# endif
-# if defined(STM32F40_41xxx) || defined (STM32F411xE)
-#  if USED_TIMERS & TIM_N(10)
-_TIM_IRQ_HANDLER2(TIM1_UP_TIM10_IRQHandler, 1, 10);  // both timers are in use
-#  else
-_TIM_IRQ_HANDLER(TIM1_UP_TIM10_IRQHandler, 1);     // timer10 is not used
-#  endif
-# endif
-# ifdef STM32F303xC
-#  if USED_TIMERS & TIM_N(16)
-_TIM_IRQ_HANDLER2(TIM1_UP_TIM16_IRQHandler, 1, 16);  // both timers are in use
-#  else
-_TIM_IRQ_HANDLER(TIM1_UP_TIM16_IRQHandler, 1);       // timer16 is not used
-#  endif
-# endif
-#endif
-#if USED_TIMERS & TIM_N(2)
-_TIM_IRQ_HANDLER(TIM2_IRQHandler, 2);
-#endif
-#if USED_TIMERS & TIM_N(3)
-_TIM_IRQ_HANDLER(TIM3_IRQHandler, 3);
-#endif
-#if USED_TIMERS & TIM_N(4)
-_TIM_IRQ_HANDLER(TIM4_IRQHandler, 4);
-#endif
-#if USED_TIMERS & TIM_N(5)
-_TIM_IRQ_HANDLER(TIM5_IRQHandler, 5);
-#endif
-#if USED_TIMERS & TIM_N(8)
-_TIM_IRQ_HANDLER(TIM8_CC_IRQHandler, 8);
-# if defined(STM32F10X_XL)
-_TIM_IRQ_HANDLER(TIM8_UP_TIM13_IRQHandler, 8);
-# else  // f10x_hd, f30x
-_TIM_IRQ_HANDLER(TIM8_UP_IRQHandler, 8);
-# endif
-# if defined(STM32F40_41xxx)
-#  if USED_TIMERS & TIM_N(13)
-_TIM_IRQ_HANDLER2(TIM8_UP_TIM13_IRQHandler, 8, 13);  // both timers are in use
-#  else
-_TIM_IRQ_HANDLER(TIM8_UP_TIM13_IRQHandler, 8);     // timer13 is not used
-#  endif
-# endif
-# if defined (STM32F411xE)
+#if defined(STM32F10X)
+
+_TIM_IRQ_HANDLER(TIM1_CC_IRQ, 1);
+_TIM_IRQ_HANDLER(TIM1_UP_IRQ, 1);
+_TIM_IRQ_HANDLER(TIM2_IRQ, 2);
+_TIM_IRQ_HANDLER(TIM3_IRQ, 3);
+_TIM_IRQ_HANDLER(TIM4_IRQ, 4);
+
+#elif defined(STM32F10X_XL) // Who uses this?
```

Comment:
STM32F10X_XL is probably defined together with STM32F10X, so this branch can't trigger anyway